### PR TITLE
BaSP-TG-68: Super Admins useForm.

### PR DESCRIPTION
**Pull request by elias-92**
-Update inputs.
-Create validations file.
-Add Reset function to clear the fields.
-Change relative paths to absolute paths.
-Implement the loader and error components.

### CODE REVIEW

File: src/Components/SuperAdmins/Form/validations.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Joi from 'joi';
+
+export const schema = Joi.object({
+  name: Joi.string()
+    .min(3)
+    .max(50)
```
Comment (LucianaPrat) [inline src/Components/SuperAdmins/Form/validations.js:6]: the maximum must be a smaller number, it should be 20

File: src/Components/SuperAdmins/Form/validations.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    .pattern(/^([^0-9]*)$/i)
+    .messages({
+      'string.pattern.base': 'Must contain only letters',
+      'string.min': 'Must have at least 3 characters',
+      'string.max': 'Must contain less than 50 characters'
+    }),
+  lastName: Joi.string()
+    .min(3)
+    .max(50)
```
Comment (LucianaPrat) [inline src/Components/SuperAdmins/Form/validations.js:15]: the maximum must be a smaller number, it should be 20

File: src/Components/SuperAdmins/Form/index.js [hunk 3]
```diff
@@ -1,60 +1,44 @@
...
-    password: ''
+  const [feedback, setFeedback] = useState({ name: '', lastName: '' });
+  const [selectedSuperAdmin] = useState(history.location.state);
+  const {
+    handleSubmit,
+    register,
+    reset,
+    formState: { errors }
+  } = useForm({
+    mode: 'onChange',
```
Comment (GuidoG95) [inline src/Components/SuperAdmins/Form/index.js:27]: here we could use 'all' method so it still validates the field if onBlur as well as onChange

File: src/Components/SuperAdmins/Form/validations.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+export const schema = Joi.object({
+  name: Joi.string()
+    .min(3)
+    .max(20)
+    .pattern(/^([^0-9]*)$/i)
+    .messages({
+      'string.pattern.base': 'Must contain only letters',
+      'string.min': 'Must have at least 3 characters',
+      'string.max': 'Must contain less than 20 characters'
```
Comment (GuidoG95) [inline src/Components/SuperAdmins/Form/validations.js:11]: in all validation messages you could use 'string.empty' to show a custom message when the field is empty

**Review by GuidoG95 — APPROVED** [review]

File: src/Components/SuperAdmins/Form/index.js [hunk 5]
```diff
@@ -1,118 +1,116 @@
...
     }
   };
 
   return (
     <section className={styles.container}>
-      <Form onSubmit={onSubmit} title={titleForm}>
+      <Form
+        onSubmit={handleSubmit(onSubmit)}
+        title={selectedSuperAdmin ? 'Edit Super Admin' : 'Add Super Admin'}
+      >
```
Comment (Baldizan) [inline src/Components/SuperAdmins/Form/index.js:67]: You could add a noValidate={!isValid} to make the form non-submittable when one or more inputs are invalid.

You need to set first, on line 25:
`formState: { errors, isValid }`

**Review by SantiagoGonzalez97 — APPROVED** [review]

**Pull request merged**